lib/src/react16_suggestors/comment_remover.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:codemod/codemod.dart';
+import 'package:over_react_codemod/src/react16_suggestors/react16_utilities.dart';
+
+/// Suggestor that removes specific comments from files by using a `List` of
+/// `RegExp`'s to identify comments that should be removed.
+class CommentRemover extends GeneralizingAstVisitor
+    with AstVisitingSuggestorMixin
+    implements Suggestor {
+  final String startString;
+  final String endString;
+
+  CommentRemover(this.startString, this.endString);
+
+  @override
+  visitCompilationUnit(CompilationUnit node) {
+    super.visitCompilationUnit(node);
+
+    int startingOffset;
+    int endingOffset;
+
+    for (var comment in allComments(node.root.beginToken)) {
+      if (comment != null) {
+        final commentText = sourceFile.getText(comment.offset, comment.end);
+
+        if (commentText.contains(RegExp(startString)) &&
+            startingOffset == null) {
+          startingOffset = comment.offset;
+        }
+
+        if (commentText.contains(RegExp(endString)) &&
+            startingOffset != null &&
+            endingOffset == null) {
+          endingOffset = comment.end;
+        }
+      }
+
+      if (startingOffset != null && endingOffset != null) {
+        try {
+          yieldPatch(startingOffset, endingOffset, '');
+          startingOffset = null;
+          endingOffset = null;
+        } catch (e, st) {
+          throw StateError('$e\n$st');
```

Comment:
Why is a StateError being thrown here? As a result of overlapping patches? This could use a comment, and potentially a more helpful error message.

Reply:
This is actually similar to [the comment above](https://github.com/Workiva/over_react_codemod/pull/57#discussion_r330644902) - there were errors getting thrown so I included the try/catch to debug easier.

 I left it deliberately thinking it'd be nice to have if it ever did hit a case that confused it, but I didn't actually have a specific case in mind that I was weary of. That's why I chose a `StateError` - it seemed the most applicable to a general unknown failure. Does that cause more confusion than good?

Reply:
I think wrapping it in `StateError` doesn't help much; if there's an exception, we should just let it bubble up instead of wrapping it in a generic error.

Reply:
🆒 🆒 🆒 I will remove that then!